src/Network/Network/AzureFirewall/NatRule/NewAzureFirewallNatRuleCommand.cs

```diff
@@ -90,6 +94,12 @@ public override void Execute()
             // Add some validation based on the type of RuleCollection (SNAT will be supported later)
             // if (MNM.AzureFirewallNatRCActionType.Dnat.Equals(ActionType))
             {
+               // One of SourceAddress or SourceIpGroup must be present
+               if ((SourceAddress == null) && (SourceIpGroup == null))
```

Comment:
it looks like you need two parameter sets,  SourceAddress and SourceIpGroup will need to be mandatory for each.

Reply:
Fixed to make SourceAddress optional. This check is to flag when both fields are optional

Reply:
No, we do not want to use the parameter set. It lead us to complication in paramterset definition as in https://github.com/Azure/azure-powershell/pull/10668/files
We are using an approach where both SourceAddress  & SourceIpGroup are optional and using checks to make sure atleast one of these is supplied by the user

Reply:
We are using similar approach for Destination Addresses in network rule.

Reply:
the similar approach is not appropriate either, breaking change fix needed